### PR TITLE
Add Persian Language

### PR DESCRIPTION
**Pull request by telvideo**

### CODE REVIEW

**Comment by akrherz** [comment]
Why is there an unrelated addition of an ant github workflow?

**Comment by telvideo** [comment]
Hello. I added it by mistake.

**Comment by telvideo** [comment]
"I have added the Persian language."

**Comment by dwd** [comment]
Looks good to me, although I can't do more than read the Farsi badly out loud (I only have rather poor Arabic). اوپن‌فایر looks OK at least!

**Review by guusdk — CHANGES_REQUESTED** [review]
Thanks for your contribution! This looks great! I've got a couple of requests:

- For people to be able to select this translation when they setup Openfire, it needs to be added to `xmppserver/src/main/webapp/setup/index.jsp`
- For people to be able to select this language after setup, it needs to be added to `xmppserver/src/main/webapp/server-locale.jsp`
- Excuse my ignorance, but is Farsi to be displayed left-to-right or right-to-left? If it is right-to-left, you'll need to add it to this method: `org.jivesoftware.util.LocaleUtils#isLocalizedTextRTL`
- I have created a ticket in our issue tracker for this change. Can you please rewrite your commit message to make a reference to [OF-2869](https://igniterealtime.atlassian.net/browse/OF-2869)? That way, it automatically pops up in our release notes.
- Can you please squash your commits into one commit? This will remove the file that was added by mistake from history.

**Comment by dwd** [comment]
It is RTL, indeed.

File: xmppserver/src/main/webapp/server-locale.jsp [hunk 1]
```diff
@@ -261,6 +261,17 @@
                     <label for="zh_CN">Simplified Chinese (zh_CN)</label>
                 </td>
             </tr>
+             <tr>
+                <td>
+                    <input type="radio" name="localeCode" value="fa_IR" <%= ("fa_IR".equals(locale.toString()) ? "checked" : "") %> id="fa_IR" />
+                </td>
+                <td>
```
Comment (guusdk) [inline xmppserver/src/main/webapp/server-locale.jsp:268]: I _think_ you need to use `fa` or `fa_IR` everywhere (here, and in other files) for this to work. You've been consistently using `fa` so I think you want to use that here, too.

File: xmppserver/src/main/webapp/server-locale.jsp [hunk 2]
```diff
@@ -261,6 +261,17 @@
                     <label for="zh_CN">Simplified Chinese (zh_CN)</label>
                 </td>
             </tr>
+             <tr>
+                <td>
+                    <input type="radio" name="localeCode" value="fa" <%= ("fa".equals(locale.toString()) ? "checked" : "") %> id="fa" />
+                </td>
+                <td>
+                    <a href="#" onclick="document.sform.localeCode[1].checked=true; return false;"><img src="images/language_fa.gif" alt="" /></a>
```
Comment (guusdk) [inline xmppserver/src/main/webapp/server-locale.jsp:269]: I do not think you need the image at all, here (this seems to be copied from the Chinese entry). The `img` tag should probably be removed. If it _is_ needed, than the file `images/language_fa.gif` should be added to this PR.

File: xmppserver/src/main/webapp/server-locale.jsp [hunk 3]
```diff
@@ -261,6 +261,17 @@
...
+                </td>
+                <td>
+                    <label for="fa">Persian (fa_IR)</label>
```
Comment (guusdk) [inline xmppserver/src/main/webapp/server-locale.jsp:272]: `fa_IR` should be `fa`, unless the translation is specific to Iranian (in which case all `fa`'s throughout this change should be changed to `fa_IR`).

**Comment by guusdk** [comment]
Hey, I appreciate that you try to address every comment that I had, but I also see that you're struggling a bit. There is no shame in that. Would you prefer me fixing this PR for you? I'd be happy to help.

What is the most correct name for the translation that you added?

- Persian
- Persian (Iran)
- (something else)?

Would you like to be credited for this change? If so, what name would you like us to use?

**Comment by telvideo** [comment]
> Hey, I appreciate that you try to address every comment that I had, but I also see that you're struggling a bit. There is no shame in that. Would you prefer me fixing this PR for you? I'd be happy to help.
> 
> What is the most correct name for the translation that you added?
> 
> * Persian
> * Persian (Iran)
> * (something else)?
> 
> Would you like to be credited for this change? If so, what name would you like us to use?

The most accurate name is "Persian (Iran)."

My name is Akbar Azimifar. I would be happy if my name is also mentioned as a contributor.

Sorry if I might not have fully understood the issue correctly.

**Comment by guusdk** [comment]
I have now applied all of the little changes that I requested myself. That was easier than trying to explain to you what I think was needed - it should have saved us both some time!

Again, thank you for your translation! I would be grateful if you could give tomorrow's nightly build a test, to see if the translation works properly!

**Comment by akrherz** [comment]
I am unsure about backporting this as the hope is we don't have any more 4.8 series releases?

**Comment by telvideo** [comment]
> I have now applied all of the little changes that I requested myself. That was easier than trying to explain to you what I think was needed - it should have saved us both some time!
> 
> Again, thank you for your translation! I would be grateful if you could give tomorrow's nightly build a test, to see if the translation works properly!

That's right, I thought the same.

**Comment by guusdk** [comment]
I have added the Persian language to our [Transifex project page](https://explore.transifex.com/igniterealtime/), for easier future updates. Sadly, it does not show all of the translated data yet. Hopefully, that automatically synchronizes in the future.

**Comment by guusdk** [comment]
Nightly builds of Openfire that include this language are now available on http://www.igniterealtime.org/downloads/nightly_openfire.jsp

@telvideo would you mind testing one, to see if everything is in order?

**Comment by telvideo** [comment]
> Nightly builds of Openfire that include this language are now available on http://www.igniterealtime.org/downloads/nightly_openfire.jsp
> 
> @telvideo would you mind testing one, to see if everything is in order?

photo_2024-08-29_16-10-08
photo_2024-08-29_16-10-25
photo_2024-08-29_16-10-48
photo_2024-08-29_16-11-22
photo_2024-08-29_16-11-53
photo_2024-08-29_16-17-37
photo_2024-08-29_16-18-21
photo_2024-08-29_16-18-39
photo_2024-08-29_16-19-14



Hello, the installation was successful. It's good, but in some parts, the appearance has issues, and it doesn't display right-to-left correctly.

**Comment by guusdk** [comment]
Thank you. If I understand your comment correctly, then it is currently usable, but should be improved. I have created a new ticket for this in our issue tracker: https://igniterealtime.atlassian.net/browse/OF-2870

If you happen to be available to apply these changes, then you would be more than welcome to. :)

**Comment by telvideo** [comment]
> Thank you. If I understand your comment correctly, then it is currently usable, but should be improved. I have created a new ticket for this in our issue tracker: https://igniterealtime.atlassian.net/browse/OF-2870
> 
> If you happen to be available to apply these changes, then you would be more than welcome to. :)

Yes, it is usable and there are no issues, but it does need improvement.

I will definitely contribute.

Thank you.

**Pull request merged**